pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -447,17 +454,24 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 				replicas[i].Name)
 			return &status, nil
 		}
-		// Enforce the StatefulSet invariants
-		if identityMatches(set, replicas[i]) && storageMatches(set, replicas[i]) {
```

Comment:
It's been pushed down into UpdateStatefulPod, actually. We can't continue when identity & storage matches, because the ownerRefs may need to be updated as well.